app/queries/servers/file.ts

```diff
@@ -16,3 +19,18 @@ export const getFileById = async (database: Database, fileId: string) => {
         return undefined;
     }
 };
+
+export function observeCanDownloadFiles(database: Database) {
+    const enableMobileFileDownload = observeConfigBooleanValue(database, 'EnableMobileFileDownload');
+    const isLicened = observeLicenseBooleanValue(database, 'IsLicensed');
+    const hasCompliance = observeLicenseBooleanValue(database, 'Compliance');
+
+    const complianceEnabled = combineLatest([isLicened, hasCompliance]).pipe(
+        switchMap(([licensed, compliance]) => of$(licensed || compliance)),
+    );
+
+    const canDownloadFiles = combineLatest([enableMobileFileDownload, complianceEnabled]).pipe(
+        switchMap(([download, compliance]) => of$(compliance && download)),
+    );
+    return canDownloadFiles;
+}
```

Comment:
compare with the previous implementation:
```ts
const license = observeLicense(database);
const enablePublicLink = observeConfigBooleanValue(database, 'EnablePublicLink');
const enableMobileFileDownload = observeConfigBooleanValue(database, 'EnableMobileFileDownload');

const complianceDisabled = license.pipe(switchMap((l) => of$(l?.IsLicensed === 'false' || l?.Compliance === 'false')));
const canDownloadFiles = combineLatest([enableMobileFileDownload, complianceDisabled]).pipe(
    switchMap(([download, compliance]) => of$(compliance || download)),
);
```

Reply:
I believe the case for allowing downloading is that all of the following must be met.  

1. isLicensed === true
2. compliance === true
3. EnableMobileFileDownload === true